scripts/device-common-install/libexec/device_hw_validation.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+is_expected_hdd() {
+  local expected_value
+  local tolerance
+
+  expected_value="${1?ERROR variables is unset}"
+  tolerance="${2?ERROR variable is unset}"
+
+  hdd_lower_bound=$(expr "$1" - "$2")
+  hdd_upper_bound=$(expr "$1" + "$2")
+  hdd_size="$(lsblk /dev/sda | head -n 2 | tail -n 1 | awk '{print $4}' | sed 's/[A-Z]//g')"
+
+  if [ 1 -eq $(echo "$hdd_size"'>='"$hdd_lower_bound" | bc -l) ] && \
+     [ 1 -eq $(echo "$hdd_size"'<='"$hdd_upper_bound" | bc -l) ]; then
+     return 0
+  else
+    return 1
+  fi
+
+  # This point should never be reached.
+  return 1
+}
+
+is_expected_ram() {
+  local expected_value
+  local tolerance
+
+  expected_value="${1?ERROR variables is unset}"
+  tolerance="${2?ERROR variable is unset}"
+
+  ram_lower_bound=$(expr "$1" - "$2")
+  ram_upper_bound=$(expr "$1" + "$2")
+  ram_size="$(free -m --si | grep Mem: | awk '{print $2}')"
+
+  if [ 1 -eq $(echo "$ram_size"'>='"$ram_lower_bound" | bc -l) ] && \
+     [ 1 -eq $(echo "$ram_size"'<='"$ram_upper_bound" | bc -l) ]; then
+     return 0
+  else
+    return 1
+  fi
+
+  # This point should never be reached.
+  return 1
+}
+
+ensure_expected_device() {
+  local hw_config
+  local key
+  local expected_value
+  local tolerance
+  local module
+  local device
+
+  hw_config="${1?ERROR variables is unset}"
+  network_interface="${2?ERROR variable is unset}"
+  device="${3?ERROR variable is unset}"
+
+  device_mac=$(ifconfig "$network_interface" | grep ether | awk '{print $2}')
+
+  if [ null == "$(echo "$hw_config" | yq --arg key "$device_mac" '.[$key]')" ]; then
+    key="default"
+  else
+    key="$device_mac"
+  fi
+
+  expected_value="$(echo "$hw_config" | yq --arg key "$device_mac" \
```

Comment:
When you read a value back to bash, use `-j`:


### Reading a value

As json:

```bash
$ echo "{}" | jq '.my_key.sub_key = "foo"' | jq '.my_key.sub_key'
"foo"
```

Shell ready, use the `-j` flag:

```bash
$ echo "{}" | jq '.my_key.sub_key = "foo"' | jq -j '.my_key.sub_key'
foo
```